gnosis/safe/safe.py

```diff
@@ -1047,6 +1068,7 @@ def retrieve_threshold(
             block_identifier=block_identifier
         )
 
+    @cached_property
```

Comment:
nitpick: since the consumers of this function will see it as a property I'd remove the `retrieve` word.

i.e. `contract.version` looks more like a property than `contract.retrieve_version`. The fact that it actually retrieves something and caches it should be hidden to whoever calls this.

Reply:
Makes sense but maybe shouldn't be done here, what do you think if I revert the cached_property and create an issue to add the property version to the contract class in a different PR?

Reply:
> Makes sense but maybe shouldn't be done here, what do you think if I revert the cached_property and create an issue to add the property version to the contract class in a different PR? 

I'm fine with that too. 

> Also I'm concerned about what would happen if the contract is updated to new version and this property is cached.

This should be an edge case which can be worked around by creating a new instance of the class. Even though it can happen we should compare this edge case to the possible reduction in RPC calls. But we can discuss this separately if you are going to revert this change 👍

Reply:
Reverted here https://github.com/safe-global/safe-eth-py/pull/422/commits/f7cee19541ea4af81177fca1eb841114e02e0e5f

Reply:
Agree on this:

`i.e. contract.version looks more like a property than contract.retrieve_version. The fact that it actually retrieves something and caches it should be hidden to whoever calls this.`

The only thing pulling me out from caching version is that it can change and break existing services